File: 12. Rekursi/C. Komposisi Fungsi II/c_komposisi_fungsi_ii.go
package main

import (
	"fmt"
	"math"
)

func fx(A int, B int, K int, x int) int {
	if K == 1 {
		return int(math.Abs(float64(A*x + B)))
	} else {
		return int(math.Abs(float64(A*fx(A, B, K-1, x) + B)))
	}
}

func main() {
	var A, B, K, x int
	fmt.Scan(&A, &B, &K, &x)
	fmt.Println(fx(A, B, K, x))
}

File: 12. Rekursi/C. Komposisi Fungsi II/c_komposisi_fungsi_ii.java
import java.util.Scanner;

public class c_komposisi_fungsi_ii {
    public static int fx(int A, int B, int K, int x) {
        if (K == 1) {
            return Math.abs(A * x + B);
        }
        return Math.abs(A * fx(A, B, K - 1, x) + B);
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int A = scan.nextInt();
        int B = scan.nextInt();
        int K = scan.nextInt();
        int x = scan.nextInt();
        scan.close();
        System.out.println(fx(A, B, K, x));
    }
}
